src/LayerPlan.cpp

```diff
@@ -622,8 +635,9 @@ TimeMaterialEstimates GCodePlanner::computeNaiveTimeEstimates()
     return ret;
 }
 
-void GCodePlanner::processFanSpeedAndMinimalLayerTime()
+void LayerPlan::processFanSpeedAndMinimalLayerTime(Point starting_position)
 {
+    extruder_plans[0].start_position = starting_position;
```

Comment:
In the end I delted a lot of code because of hits. Including LayerPlan::computeNaiveTimeEstimates and LayerPlanTest unit test....